R/poridge.R

```diff
@@ -0,0 +1,387 @@
+  }
+
+  return(newdata)
+}
+
+#' Principal coordinate ridge regression
+#'
+#' Smooth constructor function for principal coordinate ridge regression fitted by \code{\link[mgcv]{gam}}. When the principal coordinates are defined by a relevant distance among functional predictors, this is a form of nonparametric scalar-on-function regression. Reiss et al. (2016) describe the approach and apply it to dynamic time warping distances among functional predictors.
+#'
+#' @aliases pco smooth.construct.pco.smooth.spec Predict.matrix.pco.smooth poridge
+#' @export
+#'
+#' @param object a smooth specification object, usually generated by a term of the form \code{s(dummy, bs="pco", k, xt)}; see Details.
+#' @param data a list containing just the data.
+#' @param knots IGNORED!
+#'
+#' @return An object of class \code{pco.smooth}. The resulting object has an \code{xt} element which contains details of the multidimensional scaling, which may be interesting.
+#'
+#' @section Details:
+#' The constructor is not normally called directly, but is rather used internally by \code{\link{gam}}.
+#'
+#' In a \code{\link[mgcv]{gam}} term of the above form \code{s(dummy, bs="pco", k, xt)}, 
+#' \itemize{
+#'	\item \code{dummy} is an arbitrary vector (or name of a column in \code{data}) whose length is the number of observations. This is not actually used, but is required as part of the input to \code{\link[mgcv]{s}}. Note that if multiple \code{pco} terms are used in the model, there must be multiple unique term names (e.g., "\code{dummy1}", "\code{dummy2}", etc).
+#'	\item \code{k} is the number of principal coordinates (e.g., \code{k=9} will give a 9-dimensional projection of the data).
+#'	\item \code{xt} is a list supplying the distance information, in one of two ways. (i) A matrix \code{Dmat} of distances can be supplied directly via \code{xt=list(D=Dmat,\dots)}. (ii) Alternatively, one can use \code{xt=list(realdata=\dots, dist_fn=\dots, \dots)} to specify a data matrix \code{realdata} and distance function \code{dist_fn}, whereupon a distance matrix \code{dist_fn(realdata)} is created.
+#'}
+#' The list \code{xt} also has the following optional elements:
+#' \itemize{
+#'   \item \code{add}: Passed to \code{\link{cmdscale}} when performing multidimensional scaling; for details, see the help for that function. (Default \code{FALSE}.)\cr
+#'    \item \code{fastcmd}: if \code{TRUE}, multidimensional scaling is performed by \code{\link{cmdscale_lanczos}}, which uses Lanczos iteration to eigendecompose the distance matrix; if \code{FALSE}, MDS is carried out by \code{\link{cmdscale}}.
+#' }
+#'
+#' @author David L Miller, based on code from Lan Huo and Phil Reiss
+#'
+#' @references
+#' Reiss, P. T., Miller, D. L., Wu, P.-S., and Wen-Yu Hua, W.-Y. Penalized nonparametric scalar-on-function regression via principal coordinates. Under revision. Available at \url{https://works.bepress.com/phil_reiss/42/}.
+#'
+#' @examples
+#' # a simulated example
+#' library(refund)
```

Comment:
We're using a different name for package in `/devel` than in `/master`, which causes problems here: this should load `refundDevel` not `refund`. Once we merge changes into `/master` we automatically switch out the name......

```
checking examples ... ERROR
Running examples in ‘refundDevel-Ex.R’ failed
The error most likely occurred in:
> base::assign(".ptime", proc.time(), pos = "CheckExEnv")
> ### Name: smooth.construct.pco.smooth.spec
> ### Title: Principal coordinate ridge regression
> ### Aliases: Predict.matrix.pco.smooth pco poridge
> ###   smooth.construct.pco.smooth.spec
> 
> ### ** Examples
> 
> # a simulated example
> library(refund)
Error in library(refund) : there is no package called ‘refund’
```
(see https://travis-ci.org/refunders/refund/jobs/138205498)